### PR TITLE
Cleanup and adapt

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

File: memp2p/errors.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package memp2p
 
 import "errors"
 
-// ErrNilNetwork signals that a nil was given where a memp2p.Network instance was expected
+// ErrNilNetwork signals that a nil was given where a memcommon.Network instance was expected
```
Comment (iulianpascalau) [inline memp2p/errors.go:5]: we do not have memcommon :D

File: memp2p/message.go [hunk 2]
```diff
@@ -4,10 +4,10 @@ import (
 	"encoding/binary"
 
 	"github.com/ElrondNetwork/elrond-go-core/core"
-	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go-p2p/common"
 )
 
-var _ p2p.MessageP2P = (*message)(nil)
+var _ common.MessageP2P = (*message)(nil)
```
Comment (iulianpascalau) [inline memp2p/message.go:10]: I think we can drom the p2p/memp2p package, I think it is not used anymore.
Initially was supposed to be a drop-in replacement for the netMessenger. The result was a less optimal implementation

File: .github/workflows/pr-tests.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: Tests
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply .github/workflows/pr-tests.yml:1]: added

File: memp2p/message.go (hunk 2 above)
Comment (sstanculeanu) [reply memp2p/message.go:10]: removed it

File: libp2p/p2pSigner_test.go [hunk 4]
```diff
@@ -21,6 +21,32 @@ func generatePrivateKey() *libp2pCrypto.Secp256k1PrivateKey {
...
+		assert.Nil(t, err)
+		assert.NotNil(t, sig)
+	})
+	signer := &p2pSigner{
+		privateKey: generatePrivateKey(),
+	}
+
+	sig, err := signer.Sign([]byte("payload"))
+	assert.Nil(t, err)
+	assert.NotNil(t, sig)
```
Comment (ssd04) [inline libp2p/p2pSigner_test.go:47]: i think this can be removed since it's handled in the next test

File: mock/marshallerMock.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+// ErrMockMarshaller -
+var ErrMockMarshaller = errors.New("MarshallerMock generic error")
+
+// MarshallerMock that will be used for testing
+type MarshallerMock struct {
+	Fail bool
+}
+
+// Marshal converts the input object in a slice of bytes
+func (mm MarshallerMock) Marshal(obj interface{}) ([]byte, error) {
```
Comment (ssd04) [inline mock/marshallerMock.go:17]: use maybe reference to pointer?

**Review by ssd04 — APPROVED** [review]

File: .github/workflows/pr-tests.yml [hunk 6]
```diff
@@ -25,3 +25,21 @@ jobs:
...
+        id: go
+
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Get dependencies
+        run: |
+          go get -v -t -d ./...
+      - name: Unit tests
+        run: make test-race
```
Comment (iulianpascalau) [inline .github/workflows/pr-tests.yml:45]: empty line

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**